component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/cloud/DefaultResponseMetricEventBuilder.java

```diff
@@ -76,16 +64,17 @@ public boolean validate() throws MetricReportingException {
     }
 
     @Override
-    protected Map<String, Object> buildEvent() {
+    protected Map<String, Object> buildEvent() throws MetricReportingException{
         if (!isBuilt) {
+            // util function to filter required attributes
+            eventMap = EventMapAttributeFilter.getInstance().filter(eventMap,requiredAttributes);
```

Comment:
For better readability lets use a different instance variable

Reply:
As we discussed earlier Tishan..
I kept the same reference name for consistency..
because even after filtering there are some operations done on the eventmap..
For example modifying the userAgent attribute.

Reply:
Ack